node/node.go

```diff
@@ -248,16 +248,31 @@ func (n *Node) run(ctx context.Context) (err error) {
 	}
 
 	forceCertRenewal := make(chan struct{})
+	renewCert := func() {
+		select {
+		case forceCertRenewal <- struct{}{}:
+		case <-ctx.Done():
+			return
```

Comment:
You can delete the `return`. The original purpose was to make the goroutine return. Now that this is a closure, it won't happen. But when the loop repeats, `ctx.Done()` will be checked again and it will return there.